### PR TITLE
fix mimetypes

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
relates to https://github.com/rsocket/rsocket-java/issues/336

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java [hunk 1]
```diff
@@ -99,9 +100,9 @@ static ServerRSocketFactory receive() {
   }
 
   interface MimeType<T> {
-    T mimeType(String dataMineType, String metadataMimeType);
+    T mimeType(String metadataMimeType, String dataMimeType);
```
Comment (phoad) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: Even though it's good that switching the order of these two should be fixing the current users, as the users will not notice anything they can still assume that the first parameter is dataMimeType. 
So their code will still be 
```
mimeType("datamimetype", "metadatamimetype")
```
but in reality the first parameter is assigned to metadataMimeType now.

I mean, now we force them to have a bug without they notice. So I am in favor of not changing the order of these two.

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java [hunk 2]
```diff
@@ -182,15 +183,15 @@ public ClientRSocketFactory keepAliveMissedAcks(int missedAcks) {
     }
 
     @Override
-    public ClientRSocketFactory mimeType(String dataMineType, String metadataMimeType) {
-      this.dataMineType = dataMineType;
+    public ClientRSocketFactory mimeType(String dataMimeType, String metadataMimeType) {
```
Comment (phoad) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: Now the order again doesn't match to the declaration in the interface.

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java [hunk 3]
```diff
@@ -35,7 +35,8 @@
 import reactor.core.publisher.Mono;
 
 /** Factory for creating RSocket clients and servers. */
-public interface RSocketFactory {
+public interface
```
Comment (phoad) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: nit: is it in purpose that this line became two lines?

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java (hunk 1 above)
Comment (phoad) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: As all other places use this new order, I also believe it should be ..(metadataMimeType, dataMimeType)

Comment (yschimke) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: Thanks for spotting it twice!

**Pull request merged**